esphome/components/status_led/light/status_led_light.cpp

```diff
@@ -26,17 +26,24 @@ void StatusLEDLightOutput::loop() {
 
     if (lightstate_)
       lightstate_->current_values_as_binary(&state);
+    ESP_LOGD(TAG, "Restoring light state %s", ONOFF(state));
 
-    this->pin_->digital_write(state);
+    if (this->pin_ != nullptr)
+      this->pin_->digital_write(state);
+    if (this->output_ != nullptr) {
+      if (state) {
+        this->output_->turn_on();
+      } else {
+        this->output_->turn_off();
+      }
+    }
```

Comment:
FYI you can save a couple of lines using `set_state`:
```suggestion
    if (this->output_ != nullptr)
      this->output_->set_state(state);
```